mobile_app_connector/models/mobile_writing.py

```diff
@@ -0,0 +1,118 @@
+    date_start = fields.Date(
+        readonly=True,
+        states={'new': [('readonly', False)]}
+    )
+    date_stop = fields.Date(
+        readonly=True,
+        states={'new': [('readonly', False)]}
+    )
+    is_active = fields.Boolean()
+    state = fields.Selection([
+        ('new', 'New'),
+        ('active', 'Active'),
+        ('used', 'Used')
+        ], compute='_compute_state', store=True, default='new')
+
+    template_ids = fields.Many2many('correspondence.template',
+                                    string="Templates")
+
+    print_count = fields.Integer(readonly=True)
+    position = fields.Integer()
+
+    ##########################################################################
+    #                             FIELDS METHODS                             #
+    ##########################################################################
+    @api.multi
+    @api.depends('is_active')
+    def _compute_state(self):
+        for writing in self:
+            if writing.is_active:
+                writing.state = 'active'
+            else:
+                writing.state = 'used' if writing.print_count else 'new'
+
+    @api.multi
+    @api.constrains('date_start', 'date_stop')
+    def _check_dates(self):
+        for writing in self:
+            date_start = fields.Date.from_string(writing.date_start)
+            date_stop = fields.Date.from_string(writing.date_stop)
+            if date_start and date_stop and date_stop <= date_start:
+                raise ValidationError(_("Period is not valid"))
+
+    ##########################################################################
+    #                             PUBLIC METHODS                             #
+    ##########################################################################
+    @api.model
+    def validity_cron(self):
+        today = fields.Date.today()
+        active_writings = self.search([
+            ('is_active', '=', True),
+        ])
+        current_writings = self.search([
+            ('date_start', '<=', today),
+            ('date_stop', '>=', today),
+        ])
+        without_dates_writings = self.search([
+            ('date_start', '=', None),
+            ('date_stop', '=', None),
+        ])
+        # Deactivate old stories
+        (active_writings - current_writings - without_dates_writings).write(
+            {'is_active': False})
+        # Activate current stories
+        current_writings.write({'is_active': True})
+
+    @api.multi
+    def mobile_get_templates(self):
+        actives = self.env['mobile.app.writing'].search(
+            [('is_active', '=', True)])
+
+        return [
+            x.get_json() for x in actives
+        ]
+
+    @api.multi
+    def get_json(self):
+        self.ensure_one()
+        web_base_url = \
+            self.env['ir.config_parameter'].get_param('web.external.url')
+        base_url = web_base_url + "/web/image/" + self.template_ids[0]._name \
```

Comment:
Why having a many2many relation if we use only the first relation? Wouldn't be better to change it to a many2one field?